src/embed.js

```diff
@@ -19,7 +19,16 @@ const fetchHeight = (child) => {
 
 const createWidget = (container) => {
   const id = container.dataset.petitionId;
-  const url = getPetitionEmbedUrl(id);
+  const linkElements = document.getElementsByTagName('link');
+  const linksArray = Array.prototype.slice.call(linkElements);
+  // get the canonical URL
+  let referrer = null;
+  linksArray.forEach((linkElement) => {
+    if (linkElement.rel === 'canonical' && !referrer) {
+      referrer = linkElement.href;
+    }
+  });
+  const url = getPetitionEmbedUrl(id, referrer);
```

Comment:
even IE9 supports it: http://caniuse.com/#search=querySelector (click "Show All")